### ISSUE TITLE
A --check or --dry-run option

### ISSUE THREAD

**Issue by abompard** [asottile/pyupgrade#442]
Sorry for opening a new ticket but I can't reply when you lock the other issues, and I feel like I haven't made myself clear.

> you can always `git checkout -- .`

_Originally posted by @asottile in https://github.com/asottile/pyupgrade/issues/258#issuecomment-841595699_

Yes, I know I can do that, but it will also destroy their work, unless they have thought of comitting before running tox, which is a pretty big risk to take. I could also check if git has changed files before running the pre-commit hook, but it's less than ideal because it makes the tox run flaky.

I hope I'm being more clear. Sorry for my English if that's what makes things confusing.
Basically as a first step towards adoption I'd prefer having pyupgrade give advice rather than change the code directly. It's a level of trust that takes time to build, at least in my team.

**Comment by abompard** [comment]
I could also write a wrapper that copies the entire repo to a temporary directory, runs pyupgrade, and report the changes. I may end up doing that if you don't want this option, in which case I'll post the wrapper here for others who may have a similar workflow.

**Comment by abompard** [comment]
Could you please leave this ticket unlocked so users can share ideas and solutions if they need the feature? It seems that I'm not the only one.

**Comment by asottile** [comment]
look

this is **free** software, your team, your company, you are not paying a dime for it.  maybe if your company contributed a few k a month through [sponsors](https://github.com/asottile/sponsors) I'd consider something like this

you are not maintaining it

you are not contributing anything

you do not get to choose what features I need to maintain

so no, as the many many duplicates have been closed with the same consistent answer

no
